test/ConsolueModuleLoader.test.cpp

```diff
@@ -9,44 +11,60 @@ TEST(ConsoleModuleLoader, NoModuleLoaded)
 
 	// Returns empty string if no module available
 	EXPECT_EQ("", consoleModLoader.GetModuleName());
+	EXPECT_EQ("", consoleModLoader.GetModuleDirectory());
 
 	EXPECT_EQ(0u, consoleModLoader.Count());
+
 	// Module name cannot be an empty string
 	EXPECT_FALSE(consoleModLoader.IsModuleLoaded(""));
 	EXPECT_FALSE(consoleModLoader.IsModuleLoaded("TEST"));
+
+	// No module present to load, functions should return without doing anything
+	EXPECT_NO_THROW(consoleModLoader.LoadModule());
+	EXPECT_NO_THROW(consoleModLoader.RunModule());
+	EXPECT_NO_THROW(consoleModLoader.UnloadModule());
 }
 
-TEST(ConsoleModuleLoader, GetModuleDirectory)
+TEST(ConsoleModuleLoader, ModuleWithoutDLL)
 {
-	const std::string testModule("TestModule");
-	ConsoleModuleLoader consoleModLoader(testModule);
+	const auto moduleName("NoDllTest");
+	ConsoleModuleLoader consoleModLoader(moduleName);
 
-	EXPECT_NE("", consoleModLoader.GetModuleDirectory());
-	EXPECT_EQ(testModule, consoleModLoader.GetModuleDirectory());
-}
+	const auto moduleDirectory = fs::path(GetGameDirectory()) / moduleName;
+	EXPECT_EQ(moduleDirectory, consoleModLoader.GetModuleDirectory());
+	EXPECT_EQ(ToLower(moduleName), consoleModLoader.GetModuleName());
 
-TEST(ConsoleModuleLoader, GetModuleName)
-{
-	const std::string testModule("testmodule");
-	ConsoleModuleLoader consoleModLoader(testModule);
+	EXPECT_TRUE(consoleModLoader.IsModuleLoaded(moduleName));
+	EXPECT_FALSE(consoleModLoader.IsModuleLoaded(""));
+	EXPECT_FALSE(consoleModLoader.IsModuleLoaded("UnknownModule"));
+
+	EXPECT_EQ(1u, consoleModLoader.Count());
 
-	EXPECT_NE("", consoleModLoader.GetModuleName());
-	EXPECT_EQ(testModule, consoleModLoader.GetModuleName());
+	// No DLL to load, but will attempt to set memory addresses
+	EXPECT_NO_THROW(consoleModLoader.LoadModule());
+	EXPECT_NO_THROW(consoleModLoader.RunModule());
+	EXPECT_NO_THROW(consoleModLoader.UnloadModule());
```

Comment:
We could add a EnableTestEnvironment function to op2internal. This could disable modal dialogs and disable memory patching simultaneously then. 

I step debugged through the code and op2memory passed out a failure and said it could not edit the locations.

Reply:
I'm not sure how reliable that failure will be. The code would likely fail gracefully if the address was not a mapped address. In which case, failure might depend on module load address and size. As the test module grows, that size aspect will change. If ASLR is turned on, the module load address could also change.

Good that you stepped through the code though.

We may need to revisit this section of code in the context of unit testing. Your idea of disabling memory patching and modal dialogs together should work.

Reply:
I'll take a stab at disabling before merging.

Reply:
To be clear, I was thinking about commenting out those few lines from the unit tests, rather than making deep changes to the library code. We can certainly look into deeper changes, though that could be deferred until later.

Reply:
Memory patching is now disabled when in a unit test environment